modules/document_repository/jsx/uploadForm.js

```diff
@@ -144,6 +144,18 @@ class DocUploadForm extends Component {
               onUserInput={this.setFormData}
               value={this.state.formData.comments}
             />
+            {
+              loris.userHasPermission('document_repository_hidden') ?
```

Comment:
You can modify this to use reacts [in-line conditional rendering](https://reactjs.org/docs/conditional-rendering.html#inline-if-with-logical--operator) instead of ternary returning null.